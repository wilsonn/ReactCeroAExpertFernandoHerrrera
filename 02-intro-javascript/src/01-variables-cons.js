//var name = "pepe";
//let name = "pepe";
//name= "Andres";//aparece deprecate porque es palabra clave en JS

let firtsname = "pepe";
firtsname= "Andres";

const lastname = "Doe";
//lastname = "Roe";

const condicion = true;

if (condicion) {
  let name = "Ale";
  console.log(name);
  const lastname="Ale";
  console.log(lastname);
}
//console.log('Hola mundo!!! ' + firtsname + ' ' +lastname);
console.log(`Hola mundo!!! ${firtsname} - ${lastname}`);
