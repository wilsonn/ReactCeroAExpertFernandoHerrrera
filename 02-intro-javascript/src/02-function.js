//function sayHello(){
//function sayHello(name){
//function sayHello(name = 'Pepe'){
//function sayHello(name = 'Pepe', age = 0){  
  //console.log("Hola Mundo function!");
//  const greeting = `Hola mundo function! ${name} edad ${age}`;
//  return greeting;
//}

//const sayHello = (name = 'Pepe', age = 0) => {  
//  const greeting = `Hola mundo function! ${name} edad ${age}`;
//  return greeting;
//}

const sayHello = (name = 'Pepe', age = 0) => `Hola mundo function! ${name} edad ${age}`;
const add = (a = 0, b = 0) => a + b;
//const result = sayHello();
console.log(sayHello('Andres', 10));
console.log(add(10, 5))