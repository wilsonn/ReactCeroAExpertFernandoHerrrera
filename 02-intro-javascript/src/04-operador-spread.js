const invoice = {
  id: 10,
  name: 'Compras de oficina',
  date: new Date(),
  cliente: {
    id: 2,
    name: 'Jhon',
    lastname: 'Doe',
    age: 20,
  },
  items: [
    {
      producto: 'keyboard',
      price:399,
      quantity: 2,
    },
    {
      producto: "mouse",
      price: 200,
      quantity: 1,
    },
    {
      producto: "paper",
      price: 100,
      quantity: 10,
    }
  ],
  //total: 1000,
  total: function(){
    let total = 0;
    this.items.forEach(item => {
      total = total + item.price * item.quantity;
    })
    return total;
  },
  greeting: function(){
    return `Hola ${this.cliente.name}`
  }
};

//const invoice2 = invoice;
const invoice2 = {...invoice};
const result = invoice == invoice2;

if (result) {
  console.log(result);
} else {
  console.log('No son iguales');
}

invoice2.id = 20;
console.log(invoice.id);
console.log(invoice2.id);


invoice.cliente.name = 'Pepe';
//invoice.total = 5000;
console.log(invoice)

const greeting = invoice.greeting();
console.log(greeting);

console.log('Total: ' + invoice.total());