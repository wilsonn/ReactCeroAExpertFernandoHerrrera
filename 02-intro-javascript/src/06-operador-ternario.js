const average = 5.9;
const status2 = ( average >= 5.5) ? 'Aprobado' : 'Rechazado';

console.log(`Resultado: ${status2}`);

let max;

const a = 5;
const b = 8
const c = 3;

max = a > b ? a : b;
max = max > c ? max : c;

console.log(`El numero mayor es: ${max}`);