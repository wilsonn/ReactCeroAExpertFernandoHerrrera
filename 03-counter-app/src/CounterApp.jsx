import { useState } from 'react'
import PropTypes from 'prop-types' 

const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value );
  
    const handlerAdd = () => {
        setCounter(counter + 1);
    }  

    const handlerSustract = () => {
        setCounter(counter - 1);
    } 
    
    const handlerReset = () => {
        setCounter(value);
    } 

  return (
    <>
     <h1>CounterApp</h1> 
     <h2>{ counter }</h2>

     <button onClick = { handlerAdd }>
        +1
     </button>
     <button onClick = { handlerSustract }>
        -1
     </button>
     <button onClick = { handlerReset }>
        Reset
     </button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp
