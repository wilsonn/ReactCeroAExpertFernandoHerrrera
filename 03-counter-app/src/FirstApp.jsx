import PropTypes from 'prop-types'

const newMessage = {
  message: 'Hola Mundo',
  title: 'Fernando'
};

export const FirstApp = ( { title, subtitle, name } ) => {
  //console.log(props);

  return (
    <>
      <h1>{ title }</h1>
      <code>{ JSON.stringify(newMessage) }</code>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
  name: PropTypes.string
}

FirstApp.defaultProps = {
  title: "No hay titulo",
  subtitle: "No hay subtitulo",
  name: "Fernando Herrera"
}
