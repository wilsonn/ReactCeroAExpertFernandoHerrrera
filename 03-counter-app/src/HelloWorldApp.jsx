export const HelloWorldApp = () => {
  return (
    <div>
      <h1>Hello World App</h1>
    </div>
  )
}
