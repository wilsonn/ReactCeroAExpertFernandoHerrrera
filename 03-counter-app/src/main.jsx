import React  from "react";
import ReactDOM from "react-dom/client";

import { HelloWorldApp } from "./HelloWorldApp"
import { FirstApp } from "./FirstApp";

import './styles.css';

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <FirstApp title={123} subtitle="Estoy aprendiendo React"/>
    </React.StrictMode>
);