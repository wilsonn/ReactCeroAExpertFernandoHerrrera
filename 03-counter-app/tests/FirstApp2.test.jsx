import { FirstApp } from "../src/FirstApp";
import { render, screen } from '@testing-library/react'

describe('Pruebas en <FirstApp />', () => {
    const subtitle = 56;
    const title = 'Mi componente';
    
    test('debe de hacer match con el snapshot', () => {

        const { container } = render( <FirstApp subtitle={subtitle} title={title} />);

        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar el mensaje "Mi Componente"', () => {
        render( <FirstApp title={title} subtitle={subtitle} />);

        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('debe mostrar titulo en h1', () => {
        render( <FirstApp title={title} subtitle={subtitle}/>);

        expect(screen.getByRole('heading', {level: 1}).innerHTML).toBe( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render( <FirstApp subtitle={subtitle} title={title} />);

        expect( screen.getByText(subtitle) ).toBeTruthy();
    });
});