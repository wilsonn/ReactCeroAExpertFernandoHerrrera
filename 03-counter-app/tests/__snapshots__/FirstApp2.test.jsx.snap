// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Mi componente
  </h1>
  <p>
    56
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
